interfaces/builtin/pwm.go

```diff
@@ -0,0 +1,153 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/systemd"
+	"github.com/snapcore/snapd/logger"
+	"github.com/snapcore/snapd/snap"
+)
+
+// https://www.kernel.org/doc/Documentation/pwm.txt
+const pwmSummary = `allows access to specifc PWM channel`
+
+const pwmBaseDeclarationSlots = `
+  pwm:
+    allow-installation:
+      slot-snap-type:
+        - core
+        - gadget
+    deny-auto-connection: true
+`
+
+var pwmSysfsPwmChipBase = "/sys/class/pwm/pwmchip%d"
+
+// pwmInterface type
+type pwmInterface struct{}
+
+// String returns the same value as Name().
+func (iface *pwmInterface) String() string {
+	return iface.Name()
+}
+
+// Name of the pwmInterface
+func (iface *pwmInterface) Name() string {
+	return "pwm"
+}
+
+func (iface *pwmInterface) StaticInfo() interfaces.StaticInfo {
+	return interfaces.StaticInfo{
+		Summary:              pwmSummary,
+		BaseDeclarationSlots: pwmBaseDeclarationSlots,
+	}
+}
```

Comment:
these things can all be more easily specified by embedded the `commonInterface` inside the `pwmInterface` like was done for the docker interface: 

https://github.com/snapcore/snapd/blob/5831ee159bba3c432dab43875206355089678281/interfaces/builtin/docker_support.go#L651-L653

and

https://github.com/snapcore/snapd/blob/5831ee159bba3c432dab43875206355089678281/interfaces/builtin/docker_support.go#L706-L719

Reply:
Thanks for pointing it out. Let me update this accordingly.

Reply:
Applied commonInterface for PWM now! Please let me if something wrong.